shellpython/preprocessor.py

```diff
@@ -76,12 +76,18 @@ def _get_username():
 
     :return: The name of current user
     """
-    return getpass.getuser()
+    try:
+        n = getpass.getuser()
+        return n
+    except:
+    #returns temporary name. eg: temp1232433
+        from time import time
+        return 'temp'+str(int(time))
     # TODO: what if function does not work
     # TODO: see whether getpass is available everywhere
 
 
-def _translate_to_temp_path(path):
+def _translate_to_temp_path(path, getFolderPath=False):
```

Comment:
I personally against when you change interface of function in terms of the number of values returned because of set or unset parameter. I think interface should always be stable, even if it is dynamic language.